Yosemite/Yosemite/Stores/CardPresentPaymentStore.swift

```diff
@@ -196,11 +196,8 @@ private extension CardPresentPaymentStore {
     func checkForCardReaderUpdate(onCompletion: @escaping (Result<CardReaderSoftwareUpdate?, Error>) -> Void) {
         cardReaderService.checkForUpdate()
             .subscribe(Subscribers.Sink { value in
-                switch value {
-                case .failure(let error):
+                if case .failure(let error) = value {
                     onCompletion(.failure(error))
-                case .finished:
```

Comment:
@ctarda - I need to understand this finished better - it causes the publisher to fire twice when an update is available, once with the update and once with nil

for now, I've removed it, but I'll need to add back something in this PR to properly handle when no error and no update is available - see https://github.com/stripe/stripe-terminal-ios/blob/8462926231471e283eb745ac2362aacd32148b5a/docs/1.4.0/Classes/SCPTerminal.html#L1697-L1699

Were you hoping to have finished cover the no-error-no-update situation basically (with the unexpected double publishing I presume)? Or was there another reason?

Reply:
It's good to know when the subscriber has finished / the promise offered by the service has been fulfilled? Considering the pending software update is retained by the service, can it be published when the publisher finishes?

Related to that, what would be the problem if the completion block is fired twice, considering the second time it would be fired it would pass a nil as a parameter?

Reply:
> What would be the problem if the completion block is fired twice, considering the second time it would be fired it would pass a nil as a parameter?

We have three states to watch for - 1) success-update when there is an update, 2) success-nil if there is no update, 3) failure - an error occurred while trying to check for an update

Calling the completion with the update object and then again with nil when an update is available will makes the consuming logic weird.

I'll propose a fix in this PR to make the completion fire once for each of the three cases.

Reply:
> Calling the completion with the update object and then again with nil when an update is available will make the consuming logic weird.

I do not see it that way. The client can guard against a value of the software update object and exit early if it is nil.

Reply:
Take a look at the changes now that unit tests have been added. It means that when the completion is called (once) it will have the correct state: has-an-update-available, doesn't-have-an-update-available or error-checking-for-an-update

Reply:
Summarizing our offline discussion - I updated the store's handling of the update future to avoid the double cancellation and follow the Future example from raywenderlich's Combine textbook in https://github.com/woocommerce/woocommerce-ios/pull/4844/commits/7a2599d601b61a8eda1a214d5c77b26eb00b4d38

I also added a note explaining why an explicit call to `cancel()` is not needed in https://github.com/woocommerce/woocommerce-ios/pull/4844/commits/c8450c9505979ed7ed500c81613d9812206b07a9